File: angular/src/app/models/actualite.ts

import { Animal } from "./animal";
import { Ville } from "./ville";

export class Actualite {
    public get animal(): Animal {
        return this._animal;
    }
    public set animal(value: Animal) {
        this._animal = value;
    }

    public get actu_timestamp(): Date {
        return this._actu_timestamp;
    }
    public set actu_timestamp(value: Date) {
        this._actu_timestamp = value;
    }
    
    public get ville(): Ville {
        return this._ville;
    }
    public set ville(value: Ville) {
        this._ville = value;
    }

    public get actu_coordonneesgps(): string {
        return this._actu_coordonneesgps;
    }
    public set actu_coordonneesgps(value: string) {
        this._actu_coordonneesgps = value;
    }
    
    public get actu_isprivate(): boolean {
        return this._actu_isprivate;
    }
    public set actu_isprivate(value: boolean) {
        this._actu_isprivate = value;
    }

    public get actu_description(): string {
        return this._actu_description;
    }
    public set actu_description(value: string) {
        this._actu_description = value;
    }

    public get actu_id(): number {
        return this._actu_id;
    }
    public set actu_id(value: number) {
        this._actu_id = value;
    }

    constructor(private _actu_id: number,
        private _actu_description: string,
        private _actu_isprivate: boolean,
        private _actu_coordonneesgps: string,
        private _actu_timestamp: Date,
        private _animal: Animal,
        private _ville: Ville
        ) { }
}

File: spring/src/main/java/fr/formation/model/Actualite.java
package fr.formation.model;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonView;

import fr.formation.api.Views;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;

@Entity
@Table(name = "actualite")
public class Actualite {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "actu_id")
	@JsonView(Views.Common.class)
    private int actu_id;

	@Column(name = "actu_timestamp", nullable = false)
    private LocalDateTime actu_timestamp;

	@Column(name = "actu_coordonneesgps", length = 50)
    private String actu_coordonneesgps;

	@Column(name = "actu_description", length = 500)
	@NotBlank
	@JsonView(Views.Actualite.class)
    private String actu_description;

	@Column(name = "actu_isprivate", nullable = false)
	@JsonView(Views.Actualite.class)
    private boolean actu_isprivate;

//  private JSONB actu_contactsIdentifies;
    
	@ManyToOne
	@JoinColumn(name = "actu_animal_id")
	@JsonView(Views.Actualite.class)
	private Animal animal;

	@ManyToOne
	@JoinColumn(name = "actu_ville_id")
	@JsonView(Views.Actualite.class)
	private Ville ville;

	public int getActu_id() {
		return actu_id;
	}

	public void setActu_id(int actu_id) {
		this.actu_id = actu_id;
	}

	public LocalDateTime getActu_timestamp() {
		return actu_timestamp;
	}

	public void setActu_timestamp(LocalDateTime actu_timestamp) {
		this.actu_timestamp = actu_timestamp;
	}
	
	public String getActu_coordonneesgps() {
		return actu_coordonneesgps;
	}

	public void setActu_coordonneesgps(String actu_coordonneesgps) {
		this.actu_coordonneesgps = actu_coordonneesgps;
	}

	public String getActu_description() {
		return actu_description;
	}

	public void setActu_description(String actu_description) {
		this.actu_description = actu_description;
	}

	public Boolean getActu_isprivate() {
		return actu_isprivate;
	}

	public void setActu_isprivate(boolean actu_isprivate) {
		this.actu_isprivate = actu_isprivate;
	}

	public Animal getAnimal() {
		return animal;
	}

	public void setAnimal(Animal animal) {
		this.animal = animal;
	}

	public Ville getVille() {
		return ville;
	}

	public void setVille(Ville ville) {
		this.ville = ville;
	}

	@Override
	public String toString() {
		return "Actu#" + this.actu_id + " de " + this.animal.getPseudo() + " - Private = "  + this.getActu_isprivate() + " - " + this.getActu_timestamp().toLocalDate() + " à " + this.getActu_timestamp().toLocalTime() + ", " + this.getActu_description();
	}
}
